Arm/Exec.lean

```diff
@@ -166,3 +166,13 @@ theorem run_onestep (s s': ArmState) (n : Nat) (h_nonneg : 0 < n):
   · cases h_nonneg
   · rename_i n
     simp [run]
+
+/-- helper lemma for automation -/
+theorem stepi_eq_of_fetch_inst_of_decode_raw_inst
+    (s : ArmState) (addr : BitVec 64) (rawInst : BitVec 32) (inst : ArmInst)
+    (h_err    : read_err s = .None)
+    (h_pc     : read_pc s = addr)
+    (h_fetch  : fetch_inst addr s = some rawInst)
+    (h_decode : decode_raw_inst rawInst = some inst) :
+    stepi s = exec_inst inst s := by
+  simp only [stepi, h_err, h_pc, h_fetch, h_decode]
```

Comment:
So the `read_err`/`read_pc` won't show up in the goal state because of the way this lemma is used, but you're right that it's cleaner to just use `r` if it doesn't matter, so I've changed it